packages/react-native-tracer-provider/android/src/main/java/io/embrace/tracerprovider/TracerProviderModule.kt

```diff
@@ -0,0 +1,288 @@
+                    val array = map.getArray(key)
+                    val arrayType = array?.getType(0);
+                    when (arrayType) {
+                        ReadableType.Boolean -> builder.put(AttributeKey.booleanArrayKey(key), booleanListFromReadableArray(array))
+                        ReadableType.Number -> builder.put(AttributeKey.doubleArrayKey(key), doubleListFromReadableArray(array))
+                        ReadableType.String -> builder.put(AttributeKey.stringArrayKey(key), stringListFromReadableArray(array))
+                        else -> continue
+                    }
+                }
+                else -> continue
+            }
+        }
+
+
+        return builder.build();
+    }
+
+    private fun spanContextFromReadableMap(map: ReadableMap) : SpanContext {
+        val traceID = map.getString(SPAN_CONTEXT_TRACE_ID_KEY) ?: ""
+        val spanID = map.getString(SPAN_CONTEXT_SPAN_ID_KEY) ?: ""
+
+        val spanContext = SpanContext.create(
+            traceID,
+            spanID,
+            TraceFlags.getDefault(),
+            TraceState.getDefault())
+
+        return spanContext
+    }
+
+    private fun spanContextToWritableMap(spanContext: SpanContext?) : WritableMap{
+        val map = this.writableMapBuilder.build()
+
+        map.putString(SPAN_CONTEXT_TRACE_ID_KEY, spanContext?.traceId ?: "")
+        map.putString(SPAN_CONTEXT_SPAN_ID_KEY, spanContext?.spanId ?: "")
+
+        return map
+    }
+
+    init {
+        // TODO replace with Embrace OTEL provider
+        tracerProvider = SdkTracerProvider.builder()
+            .addSpanProcessor(SimpleSpanProcessor.create(LoggingSpanExporter.create()))
+            .build()
+         writableMapBuilder = WritableNativeMapBuilder()
+    }
+
+    /**
+     * Exposed for unit testing to allow a different trace provider and writableMapBuilder to be injected
+     */
+    constructor(
+        reactContext: ReactApplicationContext,
+        tracerProvider: TracerProvider,
+        writableMapBuilder: WritableMapBuilder
+    ) : this(reactContext) {
+        this.tracerProvider = tracerProvider
+        this.writableMapBuilder = writableMapBuilder
+    }
+
+    /**
+     * Methods to allow the JS side to conform to @opentelemetry-js/api
+     */
+
+    @ReactMethod fun getTracer(name: String, version: String) {
+        val id = "$name $version"
+
+        if (tracers.containsKey(id)) {
+            return
+        }
+
+        tracers[id] = tracerProvider.get(name, version)
+    }
+
+    @ReactMethod fun startSpan(tracerName: String, tracerVersion: String, spanBridgeID: String,
+                               name: String, kind: String, time: Double,
+                               attributes: ReadableMap, links: ReadableArray, parentID: String,
+                               promise: Promise) {
+        val tracer = tracers["$tracerName $tracerVersion"] ?: return;
+        val spanBuilder = tracer.spanBuilder(name);
```

Comment:
Is this semi-colon necessary? It doesn't seem to match the style of everything else around it.

Reply:
nope! There's a few that I see now, just my habit from the JS side I'll remove